bin/PCSX2_keys.ini.default

```diff
@@ -80,3 +80,30 @@ GSwindow_OffsetYplus              = Alt-Ctrl-DOWN
 GSwindow_OffsetXminus             = Alt-Ctrl-LEFT
 GSwindow_OffsetXplus              = Alt-Ctrl-RIGHT
 GSwindow_OffsetReset              = Alt-Ctrl-KP_DIVIDE
+
+# Recording Bindings
+FrameAdvance                      =	SPACE
+TogglePause                       =	Shift-P
+InputRecordingModeToggle          =	Shift-R
+
+# Save State Management
+States_SaveSlot0                  =	Shift-KP_1
+States_SaveSlot1                  =	Shift-KP_2
+States_SaveSlot2                  =	Shift-KP_3
+States_SaveSlot3                  =	Shift-KP_4
+States_SaveSlot4                  =	Shift-KP_5
+States_SaveSlot5                  =	Shift-KP_6
+States_SaveSlot6                  =	Shift-KP_7
+States_SaveSlot7                  =	Shift-KP_8
+States_SaveSlot8                  =	Shift-KP_9
+States_SaveSlot9                  =	Shift-KP_0
+States_LoadSlot0                  =	KP_1
+States_LoadSlot1                  =	KP_2
+States_LoadSlot2                  =	KP_3
+States_LoadSlot3                  =	KP_4
+States_LoadSlot4                  =	KP_5
+States_LoadSlot5                  =	KP_6
+States_LoadSlot6                  =	KP_7
+States_LoadSlot7                  =	KP_8
+States_LoadSlot8                  =	KP_9
+States_LoadSlot9                  =	KP_0
```

Comment:
Num pad will probably be used for input bindings by people that don't use a controller so it's not a good idea to bind these, same goes for space.

Do we need to have hotkey bindings for saveslots ?

Reply:
For the purposes of recording, yes they are definitely required as saving and loading save-states is done incredibly frequently (as is frame advance).  These are just the default settings, if someone were to bind numpad/space already wouldn't they already have duplicated and modified the `PCSX2_keys.ini.default` file?

I can remove the defaults and add documentation somewhere on how to bind the recording keybindings, or leave them commented out in the same file?

As an aside, is the only way to modify non-controller keybindings in PCSX2 via the ini file? I actually never noticed but looking around the UI right now I see no other way.  Would adding some sort of UI menu to change the keybindings be considered a valuable contribution?

Reply:
> As an aside, is the only way to modify non-controller keybindings in PCSX2 via the ini file? I actually never noticed but looking around the UI right now I see no other way. Would adding some sort of UI menu to change the keybindings be considered a valuable contribution?

That would be a great feature, it's finding someone willing to add it :)

> These are just the default settings, if someone were to bind numpad/space wouldn't they already have duplicated and modified the `PCSX2_keys.ini.default` file?

Lilypad has it's own ini config so hotkeys can actually become duplicated.

Reply:
If somebody's doing a TAS, I would assume it's not really any more interested into normal gaming
So... How about "enable recording tool" making lilypad use a separate .ini?

Reply:
A separate .ini file is a good idea, although I believe you meant make _pcsx2_ use a separate file?  We aren't overwriting any controller keybindings, just emulator ones (savestates and new custom ones).

In the meantime and to keep things moving, the latest commit will enable / revert the recording keybindings while preserving whatever custom overrides a user has setup (tested and everything appears to work as expected).

Setting the bindings conditionally is pretty easy, the tricky part I found was reverting and restoring the original bindings.  Right now everything is centralized in the PCSX2_Keys.ini file or in code so to pull a portion of the bindings out into their own config file, might be tricky (case in point, Lilypad's ini will overwrite).  So I'm sure there's a more elegant solution than mine, but I think I'd be best to sit on it and think about it, currently low on ideas.